File: hyperdog_gazebo_joint_cmd/hyperdog_gazebo_joint_cmd/hyperdog_gazebo_joint_controller.py
import rclpy
from rclpy.node import Node
import numpy as np
from std_msgs.msg import Float64MultiArray
from std_msgs.msg import Float32MultiArray


class hyperdog_gazebo_joint_cmd_Node(Node):
    def __init__(self):
        self.joint_angs = Float64MultiArray()
        self.rx_data = []
        self.prev_joint_angs = None
        super().__init__('hyperdog_gazebo_joint_cmd')
        self.sub_ = self.create_subscription(Float32MultiArray, 'hyperdog_jointController/commands', self.sub_callback, 30)
        self.pub_ = self.create_publisher(Float64MultiArray, 'gazebo_joint_controller/commands', 30)
        timer_period = 0.01
        # self.timerPub = self.create_timer(timer_period, callback =self.pub_callback1 )
        self.timerPub = self.create_timer(timer_period, self.pub_callback)
        
    def sub_callback(self, msg):
        self.rx_data = []
        for data in msg.data:
            self.rx_data.append(data*np.pi/180)
        self.joint_angs.data = self.rx_data
        self.pub_.publish(self.joint_angs)
            
    def pub_callback(self):
        pass
        



def main(args=None):
    rclpy.init(args=args)
    inv_kin = hyperdog_gazebo_joint_cmd_Node()
    rclpy.spin(inv_kin)
    inv_kin.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: hyperdog_gazebo_sim/src/hyperdog_gazebo_joint_controller.cpp
#include <rosidl_runtime_cpp/bounded_vector.hpp>
#include <rosidl_runtime_cpp/message_initialization.hpp>


#include <array>
#include <memory>
#include <string>
#include <vector>
#include <algorithm>

#include "chrono"
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float32_multi_array.hpp"
#include "std_msgs/msg/float64_multi_array.hpp"


// using namespace std::chrono_literals;
using std::placeholders::_1;


class HyperDogGazeboJointCtrl : public rclcpp::Node
{   
    public:
        HyperDogGazeboJointCtrl()
        : Node("Hyperdog_gazebo_joint_ctrl_node")
        {
            subscription_ = this->create_subscription<std_msgs::msg::Float32MultiArray>(
                "hyperdog_jointController/commands", 30, std::bind(&HyperDogGazeboJointCtrl::topic_callback, this, _1));
            
            publisher_ = this->create_publisher<std_msgs::msg::Float64MultiArray>("gazebo_joint_controller/commands", 30);
        }
        
    
    private:
        void topic_callback(const std_msgs::msg::Float32MultiArray::SharedPtr msg_rx) const
        {   
            auto joint_angles = std_msgs::msg::Float64MultiArray();
            std::vector<double> angles[12];
            for(float ang : msg_rx->data){
                joint_angles.data.push_back(double(ang*M_PI/180));
            }
            publisher_->publish(joint_angles);
        } 
        rclcpp::Subscription<std_msgs::msg::Float32MultiArray>::SharedPtr subscription_;
        rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr publisher_;    
};

int main(int argc, char* argv[]){
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<HyperDogGazeboJointCtrl>());
    rclcpp::shutdown();
    return 0;
}


 